tensorflow/python/ops/variables.py

```diff
@@ -226,7 +226,11 @@ def __init__(self,
           expected_shape=expected_shape)
 
   def __str__(self):
-    return str(self._snapshot)
+    return self.__repr_()
```

Comment:
That's good to know! @iRapha, could you please remove the unnecessary `__str__()` method? Thanks!